lib/template/templates.c

```diff
@@ -162,6 +162,8 @@ log_template_append_format_with_context(LogTemplate *self, LogMessage **messages
           g_static_mutex_unlock(&self->arg_lock);
           break;
         }
+        default:
```

Comment:
A template should be either a function, a macro or a value. This switch case should cover all these. Here we should use g_assert_not_reached().